core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/UserUniqueIDManger.java

```diff
@@ -67,20 +68,36 @@ public User addUser(String username, String profileName, AbstractUserStoreManage
     public User getUser(String uniqueId, AbstractUserStoreManager userStoreManager)
             throws UserStoreException {
 
-        String[] usernames = userStoreManager.getUserList(UserCoreClaimConstants.USER_ID_CLAIM_URI, uniqueId, null);
+        String userName = userStoreManager.getFromUserNameCache(uniqueId);
+        if (StringUtils.isEmpty(userName)) {
 
-        if (usernames.length > 1) {
-            throw new UserStoreException("More than one user presents with the same user unique id.");
-        }
+            String[] usernames = userStoreManager.getUserList(UserCoreClaimConstants.USER_ID_CLAIM_URI, uniqueId, null);
+
+            if (usernames.length > 1) {
+                throw new UserStoreException("More than one user presents with the same user unique id.");
+            }
 
-        if (usernames.length == 0) {
-            return null;
+            if (usernames.length == 0) {
+                return null;
+            }
+            userName = usernames[0];
+            UserStore userStore = userStoreManager.getUserStoreWithID(uniqueId);
+            String[] userNames = userName.split(CarbonConstants.DOMAIN_SEPARATOR);
+            String domainFreeUserName;
+
+            if (userNames.length > 1) {
+                domainFreeUserName = userNames[1];
+            } else {
+                domainFreeUserName = userName;
+            }
```

Comment:
I guess we should be able to use the `UserCoreUtil.removeDomainFromName` here to get the `domainFreeUserName`?

Reply:
Resolved